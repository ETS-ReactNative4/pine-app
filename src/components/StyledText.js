import React, { Component } from 'react';
import { StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';
import { withTheme } from '../contexts/theme';

const styles = StyleSheet.create({
  text: {
    fontFamily: 'System',
    fontWeight: '400'
  }
});

class StyledText extends Component {
  render() {
    const { style, theme } = this.props;

    return (
      <Text style={[styles.text, theme.text, style]} numberOfLines={this.props.numberOfLines}>
        {this.props.children}
      </Text>
    );
  }
}

StyledText.propTypes = {
  style: PropTypes.any,
  numberOfLines: PropTypes.number,
  children: PropTypes.node,
  theme: PropTypes.object
};

export default withTheme(StyledText);
