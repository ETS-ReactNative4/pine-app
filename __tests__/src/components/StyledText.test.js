import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import StyledText from '../../../src/components/StyledText';

describe('StyledText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <StyledText numberOfLines={1}>
        <Text>a7c7a003-f65a-4373-b0db-dc744c4e3e9f</Text>
      </StyledText>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
