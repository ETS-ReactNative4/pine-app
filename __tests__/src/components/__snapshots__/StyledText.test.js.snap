// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText renders correctly 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "fontFamily": "System",
        "fontWeight": "400",
      },
      Object {
        "color": "#000000",
      },
      undefined,
    ]
  }
>
  <Text>
    a7c7a003-f65a-4373-b0db-dc744c4e3e9f
  </Text>
</Text>
`;
